jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -204,6 +207,68 @@ private void uriMismatchWarningCheck(IType type, List<Diagnostic> diagnostics, I
         }
     }
 
+    /**
+     * Creates an error diagnostic if there exists more than one method annotated
+     * with @OnMessage for a given message format.
+     * 
+     * @param type
+     * @param diagnostics
+     * @param unit
+     * @throws JavaModel
+     */
+    private void onMessageWSMessageFormats(IType type, List<Diagnostic> diagnostics, ICompilationUnit unit)
+            throws JavaModelException {
+        IMethod[] typeMethods = type.getMethods();
+        boolean onMessageTextUsed = false;
+        boolean onMessageBinaryUsed = false;
+        boolean onMessagePongUsed = false;
+        for (IMethod method : typeMethods) {
+            IAnnotation[] allAnnotations = method.getAnnotations();
+            for (IAnnotation annotation : allAnnotations) {
+                if (annotation.getElementName().equals(WebSocketConstants.ON_MESSAGE)) {
+                    ILocalVariable[] allParams = method.getParameters();
+                    for (ILocalVariable param : allParams) {
+                        if (!isParamPath(param)) {
+                            String signature = param.getTypeSignature();
+                            String formatSignature = signature.replace("/", ".");
+                            String resolvedTypeName = JavaModelUtil.getResolvedTypeName(formatSignature, type);
+                            if (!resolvedTypeName.equals(WebSocketConstants.SESSION_CLASS)) {
+                                WebSocketConstants.MESSAGE_FORMAT messageFormat = getMessageFormat(resolvedTypeName);
+                                boolean duplicateFound = false;
+                                switch (messageFormat) {
+                                case TEXT:
+                                    if (onMessageTextUsed) {
+                                        duplicateFound = true;
+                                    }
+                                    onMessageTextUsed = true;
+                                    break;
+                                case BINARY:
+                                    if (onMessageBinaryUsed) {
+                                        duplicateFound = true;
+                                    }
+                                    onMessageBinaryUsed = true;
+                                    break;
+                                case PONG:
+                                    if (onMessagePongUsed) {
+                                        duplicateFound = true;
+                                    }
+                                    onMessagePongUsed = true;
+                                    break;
+                                }
+                                if (duplicateFound) {
+                                    Diagnostic diagnostic = createDiagnostic(param, unit,
```

Comment:
instead of the error showing up in the parameter, would it be better to show up in the method?

Reply:
Would it be better to have it be the method name or the method annotation?